File: ios/CustomModules.m
//
//  CustomModules.m
//  CodeChallenge
//
//  Created by Adan Sández zavala on 18/02/24.
//
#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(CustomModules, NSObject)
  RCT_EXTERN_METHOD(customModuleParams:
    (NSString *) param
    callback: (RCTResponseSenderBlock)callback
  )
@end

File: ios/CustomModules.swift
//
//  CustomModules.swift
//  CodeChallenge
//
//  Created by Adan Sández zavala on 18/02/24.
//

import Foundation

@objc(CustomModules) class CustomModules: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { return true }
  @objc public func customModuleParams(
      _ param: String,
      callback: RCTResponseSenderBlock
    ) {
      callback(["Hi I'm from a native module, and this is your param:  '\(param)'"])
    }
}
